### PR TITLE
creation of contact branch

### PR DESCRIPTION
**Pull request by ekauf**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import { Route, Routes } from "react-router-dom";
 import Home from "./pages/Home/Home";
 import "./styles/main.scss";
+import Contact from "./pages/Contact/Contact";
 
 const App = () => {
   return (
     <Routes>
       <Route path="*" element={<Home />} />
+      <Route path="/Contact" element={<Contact />} />
```
Comment (DanForder) [inline src/App.jsx:10]: lowercase for consistency: "/contact"

File: src/components/button/Button.jsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import "./Button.scss";
+
+const Button = ({ buttonText, buttonContact }) => {
+  let buttonStyle = "button";
+
+  if (buttonContact) {
+    buttonStyle = "button-contact";
```
Comment (DanForder) [inline src/components/button/Button.jsx:7]: contact is a modifier, so class should be "button--contact"

File: src/components/button/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  } else {
+    buttonStyle = "button";
+  }
+
+  return <button className={buttonStyle}>{buttonText}</button>;
```
Comment (DanForder) [inline src/components/button/Button.jsx:12]: add an onClick for this generic button component

File: src/components/button/Button.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+.button {
+  background-color: $background-color;
+
+  width: 9rem;
+
+  height: 3rem;
+
+  font-size: $default-text-size;
+
+  padding: 10px 20px;
```
Comment (DanForder) [inline src/components/button/Button.scss:15]: no pixels, use responsive units

File: src/components/button/Button.scss [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  border: solid black;
```
Comment (DanForder) [inline src/components/button/Button.scss:17]: use scss variable for colours

File: src/components/button/Button.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+.button-contact {
+  margin: auto;
+  width: 12rem;
+  font-family: $text-Font-Family;
+  color: $white;
+  margin-top: 10px;
```
Comment (DanForder) [inline src/components/button/Button.scss:25]: use responsive units

File: src/components/button/Button.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  padding: 0.5rem 0.9rem;
+
+  background-color: $green-color-palette;
+
+  font-size: 3rem;
+
+  font-weight: $default-font-weight;
+
+  border-radius: 25px;
```
Comment (DanForder) [inline src/components/button/Button.scss:34]: use responsive units

File: src/components/button/Button.scss [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+  border: none;
```
Comment (DanForder) [inline src/components/button/Button.scss:36]: remove whitespace between these lines.. you can run `npm run format` to automatically do your formatting

File: src/components/labelledInputTextField/labelledInputTextField.jsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import "./labelledInputTextField.scss";
+
+const LabelledInputTextField = ({
```
Comment (DanForder) [inline src/components/labelledInputTextField/labelledInputTextField.jsx:3]: This is quite an extreme name, how about just `TextField` ?

File: src/components/labelledInputTextField/labelledInputTextField.jsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  uniqueInputFieldHtmlName,
+  inputType,
+  labelHeader,
+  inputLarge,
+}) => {
+  let inputClassName = "labelled-input-text-field__input";
+
+  if (inputLarge) {
+    inputClassName = "labelled-input-text-field__input input-large";
```
Comment (DanForder) [inline src/components/labelledInputTextField/labelledInputTextField.jsx:12]: input-large is a modifier: `labelled-input-text-field__input--large` would be the additional classname... see why it's a good idea for a shorter component name?

File: src/components/labelledInputTextField/labelledInputTextField.jsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  } else {
+    inputClassName = "labelled-input-text-field__input";
+  }
+
+  return (
+    <>
```
Comment (DanForder) [inline src/components/labelledInputTextField/labelledInputTextField.jsx:18]: no need for a fragment here, there's only one parent element: the div

File: src/components/labelledInputTextField/labelledInputTextField.jsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
+import "./labelledInputTextField.scss";
+
+const LabelledInputTextField = ({
+  uniqueInputFieldHtmlName,
```
Comment (DanForder) [inline src/components/labelledInputTextField/labelledInputTextField.jsx:4]: this prop represents an id, so perhaps just call it `id` or `uniqueId` for simplicity

File: src/pages/Contact/Contact.jsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
+import Layout from "../../components/Layout/Layout";
+import LabelledInputTextField from "../../components/labelledInputTextField/labelledInputTextField";
+import Button from "../../components/button/Button";
+import "./contact.scss";
+
+const Contact = () => {
+  return (
+    <>
```
Comment (DanForder) [inline src/pages/Contact/Contact.jsx:8]: no need for a fragment, there's only one parent element: the Layout

File: src/pages/Contact/Contact.jsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+      <Layout>
+        <main className="contact-container">
```
Comment (DanForder) [inline src/pages/Contact/Contact.jsx:10]: Using BEM in react, the parent class should be the name of the component. so this should be "contact" and everything inside it should be "contact__"

File: src/pages/Home/Home.jsx [hunk 15]
```diff
@@ -1,19 +1 @@
...
-        dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis
-        maxime tempora, temporibus beatae voluptas repellat rerum. Dignissimos,
-        necessitatibus.
-      </p>
-    </Layout>
-  );
-};
-
-export default Home;
+// plz dont merge this file  =)
```
Comment (DanForder) [inline src/pages/Home/Home.jsx:1]: revert changes

File: src/styles/_variables.scss [hunk 16]
```diff
@@ -5,6 +5,32 @@ $desktop-breakpoint: 1024px;
...
+$white: #f3f2e8;
+$black: #1f1b1b;
+
+$green-color-palette: #5bb27c;
+
+/*                          Font stuff              */
+
+// font families
+
+$header-Font-Family: "Roboto";
```
Comment (DanForder) [inline src/styles/_variables.scss:24]: lower-case-hyphenated

**Review by DanForder — CHANGES_REQUESTED** [review]

File: src/pages/Contact/Contact.jsx [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+const Contact = () => {
+  return (
+    <Layout>
+      <main className="contact">
+        <h1>Contact</h1>
+        <p>We would love to hear from you and take your feedback on board.</p>
+        <form className="contact__form">
+          <TextField
+            uniqueInputFieldName="contactName"
```
Comment (DanForder) [inline src/pages/Contact/Contact.jsx:14]: *uniqueId

File: src/components/TextField/TextField.test.jsx [hunk 18]
```diff
@@ -0,0 +1,23 @@
+import TextField from "./TextField";
+import { customRender } from "../../utils/testUtils";
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+it("should render the TextField component", () => {
+  const { container } = customRender(<TextField />);
```
Comment (DanForder) [inline src/components/TextField/TextField.test.jsx:7]: add labelheader and unique id

File: src/components/button/Button.scss [hunk 19]
```diff
@@ -0,0 +1,40 @@
+@use "../../../src/styles/reset" as *;
+@use "../../../src/styles/variables" as *;
+@use "../../styles/typography" as *;
```
Comment (DanForder) [inline src/components/button/Button.scss:3]: remove typgraphy

**Review by DanForder — CHANGES_REQUESTED** [review]

**Review by DanForder — APPROVED** [review]

**Review by katherinepv — APPROVED** [review]

**Pull request merged**